### PR TITLE
etcd3, unidling controller

### LINKED ISSUE #10773: Extended.idling and unidling unidling should work with TCP (when fully idled)
```
------------------------------
idling and unidling unidling 
  should work with TCP (when fully idled) [Conformance]
  /data/src/github.com/openshift/origin/test/extended/idling/idling.go:309

STEP: Creating a kubernetes client
Sep  1 14:31:46.435: INFO: >>> kubeConfig: /tmp/openshift/openshift/test-extended/core/openshift.local.config/master/admin.kubeconfig

STEP: Building a namespace api object
Sep  1 14:31:46.479: INFO: configPath is now "/tmp/openshift/extended-test-cli-idling-r87sd-xpgny-user.kubeconfig"
Sep  1 14:31:46.479: INFO: The user is now "extended-test-cli-idling-r87sd-xpgny-user"
Sep  1 14:31:46.479: INFO: Creating project "extended-test-cli-idling-r87sd-xpgny"
STEP: Waiting for a default service account to be provisioned in namespace
STEP: Building a namespace api object
Sep  1 14:31:47.020: INFO: configPath is now "/tmp/openshift/extended-test-cli-idling-r87sd-cvgex-user.kubeconfig"
Sep  1 14:31:47.020: INFO: The user is now "extended-test-cli-idling-r87sd-cvgex-user"
Sep  1 14:31:47.020: INFO: Creating project "extended-test-cli-idling-r87sd-cvgex"
STEP: Waiting for a default service account to be provisioned in namespace
STEP: Creating the resources
Sep  1 14:31:47.389: INFO: Running 'oc create --namespace=extended-test-cli-idling-r87sd-cvgex --config=/tmp/openshift/extended-test-cli-idling-r87sd-cvgex-user.kubeconfig -f /data/src/github.com/openshift/origin/test/extended/testdata/idling-echo-server.yaml -o name'
STEP: Creating the idling file
STEP: Waiting for the endpoints to exist
STEP: Waiting for endpoints to be up
STEP: Idling the service
Sep  1 14:32:19.027: INFO: Running 'oc idle --namespace=extended-test-cli-idling-r87sd-cvgex --config=/tmp/openshift/extended-test-cli-idling-r87sd-cvgex-user.kubeconfig --resource-names-file /tmp/openshift/openshift-extended-tests/idling-services-246870143'
STEP: Waiting for the pods to have terminated
STEP: Connecting to the service IP and checking the echo
STEP: Collecting resource usage data
Sep  1 14:32:36.302: INFO: Closed stop channel. Waiting for 1 workers
Sep  1 14:32:36.302: INFO: Closing worker for 172.18.0.100
Sep  1 14:32:36.302: INFO: Waitgroup finished.
Sep  1 14:32:36.302: INFO: Unknown output type: . Skipping.
Sep  1 14:32:36.302: INFO: Waiting up to 1m0s for all nodes to be ready
STEP: Destroying namespace "extended-test-cli-idling-r87sd-xpgny" for this suite.
STEP: Destroying namespace "extended-test-cli-idling-r87sd-cvgex" for this suite.
STEP: Cleaning up the idling file


• Failure [74.984 seconds]
idling and unidling
/data/src/github.com/openshift/origin/test/extended/idling/idling.go:466
  unidling
  /data/src/github.com/openshift/origin/test/extended/idling/idling.go:465
    should work with TCP (when fully idled) [Conformance] [It]
    /data/src/github.com/openshift/origin/test/extended/idling/idling.go:309

    Expected error:
        <*errors.errorString | 0xc82101c1e0>: {
            s: "unable to read data from echo server for service idling-echo: read tcp 172.18.0.100:55154->172.30.12.10:8675: read: connection reset by peer",
        }
        unable to read data from echo server for service idling-echo: read tcp 172.18.0.100:55154->172.30.12.10:8675: read: connection reset by peer
    not to have occurred

    /data/src/github.com/openshift/origin/test/extended/idling/idling.go:297
------------------------------
```

As seen in:
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/5823/consoleText

### PR DESCRIPTION
**Pull request by liggitt**
* changes default storage backend to etcd2 (upstream is still defaulting to etcd2, etcd3 backend has fan-out issues calling Get() when processing delete/modify watch events)
* lists events for unidling controller

Fixes #10773

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
[test]

**Comment by smarterclayton** [comment]
Switch back to json, proto with etcd2 is not supported.

File: pkg/cmd/server/start/master_args.go [hunk 1]
```diff
@@ -344,13 +344,13 @@ func (args MasterArgs) BuildSerializeableMasterConfig() (*configapi.MasterConfig
 	configapi.SetProtobufClientDefaults(config.MasterClients.OpenShiftLoopbackClientConnectionOverrides)
 	configapi.SetProtobufClientDefaults(config.MasterClients.ExternalKubernetesClientConnectionOverrides)
 
-	// Default storage backend to etcd3 with protobuf storage for our innate config when starting both
+	// Default storage backend to etcd2 with protobuf storage for our innate config when starting both
 	// Kubernetes and etcd.
 	if km := config.KubernetesMasterConfig; km != nil && config.EtcdConfig != nil {
```
Comment (smarterclayton) [inline pkg/cmd/server/start/master_args.go]: Just comment this whole section out

**Comment by mfojtik** [comment]
@liggitt I thought this PR is fixing the unidling flake, but it seems it has just occurred...

**Comment by ncdc** [comment]
It should fix at least 1 known cause of it. I'm investigating now.

**Comment by ncdc** [comment]
Yeah, this failure was different. It's a failure in JustBeforeEach, before it even gets to trying to idle/unidle:

```
• Failure in Spec Setup (JustBeforeEach) [208.838 seconds]
idling and unidling
/data/src/github.com/openshift/origin/test/extended/idling/idling.go:466
  unidling [JustBeforeEach]
  /data/src/github.com/openshift/origin/test/extended/idling/idling.go:465
    should work with TCP (when fully idled) [Conformance]
    /data/src/github.com/openshift/origin/test/extended/idling/idling.go:309

    Expected error:
        <*errors.errorString | 0xc8200df780>: {
            s: "timed out waiting for the condition",
        }
        timed out waiting for the condition
    not to have occurred

    /data/src/github.com/openshift/origin/test/extended/idling/idling.go:244
```

I'm looking to see why this happened.

**Comment by ncdc** [comment]
This failed because it timed out pulling openshift/node. I'm assuming (hoping?) we already have this image on the node, and we just need to modify the pod spec to set imagePullPolicy to IfNotPresent :-(

**Comment by mfojtik** [comment]
@ncdc I checked and the fixture has not imagePullPolicy set which AFAIK defaults to IfNotPresent?

**Comment by ncdc** [comment]
@mfojtik unless you don't specify a tag or digest, in which case it defaults to the `latest` tag, which then results in the imagePullPolicy defaulting to Always :-(

**Comment by mfojtik** [comment]
@ncdc no latest: `image: openshift/node`

**Comment by ncdc** [comment]
@mfojtik absence of tag == :latest

**Comment by liggitt** [comment]
> Switch back to json, proto with etcd2 is not supported.

done

**Comment by ncdc** [comment]
[merge]

**Pull request merged**